imap/mupdate-client.c

```diff
@@ -88,6 +89,17 @@ static struct protocol_t mupdate_protocol =
       { "Q01 LOGOUT", NULL, "Q01 " } } }
 };
 
+/* We're really only looking for an OK or NO or BAD here -- and the callback
+ * is never called in those cases.  So if the callback is called, we have
+ * an error! */
+static int mupdate_scarf_one(struct mupdate_mailboxdata *mdata __attribute__((unused)),
+                             const char *cmd,
+                             void *context __attribute__((unused)))
+{
+    syslog(LOG_ERR, "mupdate_scarf_one was called, but shouldn't be.  Command received was %s", cmd);
+    return -1;
+}
```

Comment:
Might be nice to assert in here that it is actually an error.  If someone changes the behaviour elsewhere, such that this starts getting called for OK/etc, the subsequent assertion failure will remind them to update this as well, whereas the log line might go unnoticed, and the -1 return might be hard to debug.